spec/controllers/wishlists_controller_spec.rb

```diff
@@ -1,11 +1,23 @@
 require "rails_helper"
 
 RSpec.describe WishlistsController do
-  context "when user is logged in" do
-    before { sign_in_user }
-    let(:wishlist) { create(:wishlist, user: controller.current_user) }
+  let(:wishlist) { create(:wishlist, user: controller.current_user) }
 
-    describe "GET #index" do
+  shared_examples "redirects to" do |path_helper|
+    it "redirects to the #{path_helper}" do
+      expect(response).to redirect_to(send(path_helper))
+    end
+  end
+
+  shared_examples "shows a flash message" do |flash_type|
+    it "shows a/an #{flash_type}" do
+      expect(flash[flash_type.to_s]).not_to be_nil
+    end
+  end
```

Comment:
I've commented on the exact same thing here:
https://github.com/Ragnarson-Internship2016/group-I/pull/86/files#r72648899